lib/config/options/index.ts

```diff
@@ -246,9 +246,10 @@ const options: RenovateOptions[] = [
     name: 'dryRun',
     description:
       'If enabled, perform a dry run by logging messages instead of creating/updating/deleting branches and PRs.',
-    type: 'boolean',
+    type: 'string',
     globalOnly: true,
-    default: false,
+    allowedValues: ['extract', 'lookup', 'full'],
+    default: '',
```

Comment:
I think it should be null instead of empty string

Reply:
what you mean with migration code to not break existing users??
if user inserted true or false instead of the new allowed values,
we should support this case also?

Reply:
Yes, we always try to be backwards compatible. So please check migrations in `lib/config` folder for samples